src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,118 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_EMPTY_CAPACITY = 0;
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private final T[] defaultEmptyArray;
+    private T[] elementData;
+
+    public ArrayList() {
+        defaultEmptyArray = (T[]) new Object[DEFAULT_EMPTY_CAPACITY];
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public void grow() {
```

Comment:
Private ones (methods, classes) should be after public ones.
```suggestion
    private void grow() {
```